### ISSUE TITLE
Incorrect implementation of RMVPE's model inference

### ISSUE THREAD

**Issue by yxlllc** [lars76/pitch-benchmark#3]
I'm glad to see that you recently added RMVPE to the benchmark, but its score is not quite realistic. I noticed that you are using the RMVPE checkpoint of the RVC project. This model is actually trained by me, and its implementation differs from the original RMVPE repository. You can refer to https://github.com/yxlllc/RMVPE or https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/main/infer/lib/rmvpe.py for details. 

I reviewed your implementation https://github.com/lars76/pitch-benchmark/blob/main/algorithms/rmvpe.py and found 4 main issues:

1. WINDOW_LENGTH = 2048 should be 1024 (this differs from the original repository).https://github.com/lars76/pitch-benchmark/blob/338c2fa0b51c535bcc9512c5ba807786dd5f5efc/algorithms/rmvpe.py#L21
For reference, here is the MelSpectrogram initialization code in RVC repository
https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/7ef19867780cf703841ebafb565a4e47d1ea86ff/infer/lib/rmvpe.py#L503-L505

2. My checkpoint is trained with hop_length = 160 (10ms per frame), but your pipeline uses inference with hop_length = 320 (20ms per frame). While this still produces meaningful results, the accuracy may be underestimated. Considering that your data is all 20ms per frame, I suggest starting with inference with hop_length=160 and discarding exactly half of the f0 values ​​(equivalent to downsampling) to make a fair comparison.

3. You set the threshold to 0.75; the current project uses a very low 0.03 to significantly improve noise immunity.https://github.com/lars76/pitch-benchmark/blob/338c2fa0b51c535bcc9512c5ba807786dd5f5efc/algorithms/rmvpe.py#L743-L745
For reference, here is the threhold setting code in the RVC repository
https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/7ef19867780cf703841ebafb565a4e47d1ea86ff/infer/lib/rmvpe.py#L594

4. The checkpoint in the RVC project is actually quantized to fp16, which reduces the size by half to save space and speed-up inference, although it hardly affects the accuracy. The original unquantized fp32 checkpoint is located at https://github.com/yxlllc/RMVPE/releases/tag/230917. For fair comparison, it's recommended to use this checkpoint for testing (assuming other neural network algorithms are all inferred using fp32).

Considering that you almost reimplemented the inference, I am not sure if there are other problems (I don't have time to check it line by line). I strongly suggest you use the inference script provided in my repository https://github.com/yxlllc/RMVPE/blob/main/main.py to compare the f0 curves and test them after they are completely consistent. I believe that with correct implementation the benchmark scores can be greatly improved.

**Comment by yxlllc** [comment]
https://github.com/lars76/pitch-benchmark/pull/4
Done.

**Comment by blaisewf** [comment]
you are the goat 👏😎

**Comment by lars76** [comment]
Thanks for your contribution! I reran all the benchmarks, and RMVPE is now ranked 2nd, surpassing CREPE. It achieves the best results on MIR1k and Vocadito (two human singing datasets) and, together with SPICE and SwiftF0, produces the smoothest pitch curves. Full results are in the automatically generated [benchmark report](https://github.com/lars76/pitch-benchmark/blob/main/benchmark_report.md).

I'll be checking out your repo and the paper more closely to see if I can push the algorithms further.
